tests/system/src/test/java/test/robot/javafx/scene/MouseLocationOnScreenTest.java

```diff
@@ -118,15 +118,22 @@ public void testMouseLocation() throws Exception {
      * returned by robot are same
      */
     static void validate(Robot robot, int x, int y) {
-        Assertions.assertEquals(x, (int) robot.getMouseX());
-        Assertions.assertEquals(y, (int) robot.getMouseY());
+        try {
+            Util.sleep(DELAY_TIME);
+            Assertions.assertEquals(x, (int) robot.getMouseX());
```

Comment:
this works also, though I'd simply use `if`.

A bigger question is why are we using `int` coordinates?  Wouldn't it make the test depend on the scale (and also on the screen resolution and window position)?

Should we instead use the double coordinates and `Assertions.equals(double, double, double)` ?

Reply:
I would recommend to use a for loop as below. 
```
    private static int VALIDATE_COUNT = 3;
    static void validate(Robot robot, int x, int y) {
        for (int i = 0; i < VALIDATE_COUNT; i++) {
            Util.sleep(DELAY_TIME);
            if (x == (int)robot.getMouseX() &&
                y == (int)robot.getMouseY()) {
                break;
            }
        }
        Assertions.assertEquals(x, (int) robot.getMouseX());
        Assertions.assertEquals(y, (int) robot.getMouseY());
    }
```

Reply:
Good idea. As a (minor) suggestion, I might loop "5" times before calling it a failure.

Reply:
> A bigger question is why are we using `int` coordinates? Wouldn't it make the test depend on the scale (and also on the screen resolution and window position)?

I run this frequently on my Windows laptop with 125% and have never seen a problem. Unless we do, this seems unrelated to the bug being fixed.

Reply:
I'd recommend waiting for a few short times (1, 3, 5ms seems good), and then an additional long time just in case the machine has experienced a hiccup at that very moment. There's no harm done in waiting a second for a test to fail, as that would not slow down tests in most cases.

Reply:
@Gopalora if you adopt Michael's addition to Ambarish's suggestion, I recommend leaving `DELAY_TIME` at 1, looping for 5 times, and then having one final test for 500 or 1000 msec (0.5 or 1.0 seconds).